### ISSUE TITLE
"no-unnecessary-type-assertion" flags necessary assertion `Element as HTMLElement`

### ISSUE THREAD

**Issue by aaronbeall** [palantir/tslint#3569]
### Bug Report

- __TSLint version__: ^5.8.0
- __TypeScript version__: ^2.6.2
- __Running TSLint via__: Package script: `tslint --format stylish --project tsconfig.json`

#### TypeScript code being linted

```ts
(ReactDOM.findDOMNode(this.toggle) as HTMLElement).focus()
```

with `tslint.json` configuration:

```json
{
"no-unnecessary-type-assertion": true
}
```

#### Actual behavior

> `no-unnecessary-type-assertion`  This assertion is unnecessary since it does not change the type of the expression.

#### Expected behavior

No error. `findDOMNode()` returns `Element` which has no `focus()` method, subclass `HTMLElement` does so assertion is necessary.

The `findDOMNode()` definition from `@types/react-dom` looks like this:

```ts
declare namespace ReactDOM {
    function findDOMNode<E extends Element>(instance: ReactInstance): E;
}
```
`Element` comes from `lib.es6.d.ts`.

**Comment by ajafff** [comment]
This is a duplicate of #3540
The cause and the workaround are the same for this issue.

**Comment by aaronbeall** [comment]
@ajafff Thanks for the fast response! That thread explains it all and the workaround worked nicely in my case.
